src/query_executor.c

```diff
@@ -8,16 +8,19 @@
 #include <assert.h>
 #include "query_executor.h"
 #include "graph/graph.h"
-#include "graph/node.h"
-#include "graph/graph_type.h"
+#include "graph/entities/node.h"
 #include "stores/store.h"
 #include "util/vector.h"
 #include "parser/grammar.h"
 #include "arithmetic/agg_ctx.h"
 #include "arithmetic/repository.h"
 #include "parser/parser_common.h"
 
-static void _returnClause_ExpandCollapsedNodes(RedisModuleCtx *ctx, AST_Query *ast, const char *graphName) {
+static void _returnClause_ExpandCollapsedNodes(GraphContext *gc, AST_Query *ast) {
+    /* gc can validly be NULL if the query is a CREATE or MERGE, but in those cases
```

Comment:
At some point we should support having both CREATE/MERGE and a RETURN clause.

Reply:
Agreed, it's an unnecessary limitation.